### ISSUE TITLE
Interpolate lat_lon to lon_lat and vice versa

### ISSUE THREAD

**Issue by jmilloy** [creare-com/podpac#35]
**Description**
After the Coordinate refactor, we cannot request lat_lon data from a DataSource with lon_lat native coordinates (and vice versa).

**Steps to Reproduce**
```
class B(NumpyArray):
    source = 2 * np.arange(100.)

    def get_native_coordinates(self):
        return Coordinate(lat_lon=[(-20, 20), (0, 5), 100])

b = B()
b.execute(Coordinate(lon_lat=[(-20, 20), (0, 5), 100]))
```

**Expected Behavior**
The data should be retrieved.

**Observed Behavior**
Currently a NotImplementedError is raised. Without it, an IndexError is raised in `DataSource.interpolate_point_data`.

**Comment by jmilloy** [comment]
@mlshapiro Hopefully we can avoid spending time to fix this until you get to the interpolation refactor.

**Comment by mlshapiro** [comment]
will be addressed in #7

**Comment by mlshapiro** [comment]
this will actually get fully addressed in #133. 

For now what I'm going to do is relax the "short-circuit" on the `interpolation.interpolate` method so that if the source_coordinates contains the eval_coordinates it will handle this case without actually doing any interpolation. 

See L1194 of interpolate.py
